core/mm/vm.c

```diff
@@ -1225,7 +1227,7 @@ TEE_Result vm_pa2va(const struct user_mode_ctx *uctx, paddr_t pa, void **va)
 		}
 	}
 
-	return TEE_ERROR_ACCESS_DENIED;
+	return TEE_ERROR_NO_DATA;
```

Comment:
We do want to keep this as it means that it's an invalid virtual address.

Reply:
In our case, the PA maybe be a valid address but vm_pa2va() can't resolve it because of the pager. So TEE_ERROR_ACCESS_DENIED seems not really accurate here.
That said, the sole caller of vm_pa2va() does not really care is the error is ACCESS_DENIED, NOT_SUPPORTED or NO_DATA.

Reply:
OK, then I think it would be more clear to return the `va` on success or `NULL` on failure.

Reply:
Ok, i'll update.